src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2080,10 +2079,6 @@ private function registerCacheConfiguration(array $config, ContainerBuilder $con
             $container->removeDefinition('cache.default_marshaller');
         }
 
-        if (!class_exists(DoctrineAdapter::class)) {
-            $container->removeDefinition('cache.adapter.doctrine');
-        }
-
         if (!class_exists(DoctrineDbalAdapter::class)) {
             $container->removeDefinition('cache.adapter.doctrine_dbal');
         }
```

Comment:
Since symfony/cache 5.4+ is required by the bundle, we can remove the conditional remove of DBAL adapter.

Reply:
right! removed